gensim/models/translation_matrix.py

```diff
@@ -0,0 +1,257 @@
+
+        source_space.normalize()
+        target_space.normalize()
+
+        m1 = source_space.mat[[source_space.word2index[item] for item in self.source_word], :]
+        m2 = target_space.mat[[target_space.word2index[item] for item in self.target_word], :]
+
+        return np.linalg.lstsq(m1, m2, -1)[0]
+
+    def save(self, *args, **kwargs):
+        """
+        Save the model to file but ignoring the souce_space and target_space
+        """
+        kwargs['ignore'] = kwargs.get('ignore', ['source_space', 'target_space'])
+
+        super(TranslationMatrix, self).save(*args, **kwargs)
+
+    @classmethod
+    def load(cls, *args, **kwargs):
+        """ load the pre-trained translation matrix model"""
+        model = super(TranslationMatrix, cls).load(*args, **kwargs)
+        return model
+
+    def apply_transmat(self, words_space):
+        """
+        mapping the source word vector to the target word vector using translation matrix
+        Args:
+            words_space: the Space object that constructed for those words to be translate
+
+        Returns:
+            A Space object that constructed for those mapped words
+        """
+        return Space(np.dot(words_space.mat, self.translation_matrix), words_space.index2word)
+
+    def translate(self, source_words, topn=5, additional=None, source_lang_vec=None, target_lang_vec=None):
+        """
+        translate the word from the source language to the target language, and return the topn
+        most similar words.
+         Args:
+            source_words(str/list): single word or a list of words to be translated
+            topn: return the top N similar words. By default (`topn=5`)
+            additional: defines the training algorithm. By default (`additional=None`), use standard NN retrieval.
+            Otherwise use corrected retrieval(as described in[1]), additional is an int that specify the number of
+            word to sample from the source lexicon.
+            source_lang_vec: you can specify the source language vector for translation, the default is to use
+            the model's source language vector.
+            target_lang_vec: you can specify the target language vector for retrieving the most similar word,
+            the default is to use the model's target language vector.
+        Returns:
+            A OrderedDict object, each item is (word : topn translated words)
+
+        [1] Dinu, Georgiana, Angeliki Lazaridou, and Marco Baroni. "Improving zero-shot learning by mitigating the
+        hubness problem." arXiv preprint arXiv:1412.6568 (2014).
+        """
+
+        if isinstance(source_words, string_types):
+            # pass only one word to translate
+            source_words = [source_words]
+
+        # if the language word vector not provided by user, use the model's
+        # language word vector as default
+        if source_lang_vec is None:
+            warnings.warn("the parameter source_lang_vec didn't specified,"
```

Comment:
We ignore line size limitation, please write all comment in one line (here and below). Also, use full stops and capitals where your start your sentence.